src/ruby/lib/grpc/generic/rpc_server.rb

```diff
@@ -407,7 +413,13 @@ def loop_handle_server_calls
       request_call_tag = Object.new
       until stopped?
         deadline = from_relative_time(@poll_period)
-        an_rpc = @server.request_call(@cq, request_call_tag, deadline)
+        begin
+          an_rpc = @server.request_call(@cq, request_call_tag, deadline)
+        rescue Core::CallError, RuntimeError => e
+          # can happen during server shutdown
```

Comment:
I propose to merge this as is, given its solidity, and address the issue around success == true in separate PR.
@ctiller, if that proposal is acceptable, please merge the PR.